Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -13311,6 +13311,48 @@ void SpeculativeJIT::compileStringReplace(Node* node)
     if (sample)
         m_jit.incrementSuperSamplerCount();
 
+    if (node->op() == StringReplace
+        && node->child1().useKind() == StringUse
+        && node->child2().useKind() == StringUse
+        && node->child3().useKind() == StringUse) {
+        if (JSString* replace = node->child3()->dynamicCastConstant<JSString*>(); replace && !replace->length()) {
```

Comment:
nit: I think the more canonical JSC thing to do is to have a rule in strength reduction that transforms this node into another node if the replace string is empty. Not super critical to do, but maybe a touch nicer.

Reply:
That sounds good! We are now planning to add more optimization for constant string searching case.
So, in that next patch, I'll convert it to different node in strength-reduction or abstract-interpreter, and then, we will have a specialized optimization for that node's compilation.